packer/linux/conf/cloudwatch-agent/config.json

```diff
@@ -2,6 +2,29 @@
 	"agent": {
 		"run_as_user": "root"
 	},
+	"metrics": {
+		"metrics_collected": {
+			"mem": {
+				"measurement": [
+					"mem_used_percent"
+				]
+			},
+			"disk": {
+				"measurement": [
+					"used_percent"
+				],
+				"resources": [
+					"*"
+				]
+			}
+		},
+		"append_dimensions": {
+			"ImageId": "${aws:ImageId}",
+			"InstanceId": "${aws:InstanceId}",
+			"InstanceType": "${aws:InstanceType}",
+			"AutoScalingGroupName": "${aws:AutoScalingGroupName}"
```

Comment:
for purposes of cloudwatch metric billing, each unique dimension value is considered 1 metric, so including granular dimensions like `InstanceId` or even `InstanceType` could really blow up costs for those of us with a large number of ec2 instances.

Reply:
Good point. How about having just the `AutoScalingGroupName` dimension then?
Or would there be a way to have the queue as a dimension?

Reply:
yeah i think that makes sense. not sure if it's possible to add the queue, but i think that'd be nice as well. It should be available as a tag on the instance, `BuildkiteQueue`.

Reply:
I looked into this a bit, and one way to add `BuildkiteQueue` as a dimension would be to modify the cloudwatch config file on boot using jq ([similar to docker](https://github.com/buildkite/elastic-ci-stack-for-aws/blob/19278b86f3e08cce38f144811f9c9c0c253589fd/packer/linux/conf/bin/bk-configure-docker.sh#L18)), as part of the `bk-install-elastic-stack.sh` script. You should have access to [BUILDKITE_QUEUE](https://github.com/buildkite/elastic-ci-stack-for-aws/blob/19278b86f3e08cce38f144811f9c9c0c253589fd/packer/linux/conf/bin/bk-install-elastic-stack.sh#L117) from there and then just enable and start the service on boot instead of [at AMI creation time](https://github.com/buildkite/elastic-ci-stack-for-aws/blob/d2230c5163ecdf4cc87f46c8c72af2ea02f5b11e/packer/linux/scripts/install-cloudwatch-agent.sh#L13).